flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -434,6 +441,51 @@ struct CharacterOpsBuilderImpl {
 
     return {substringRef, substringLen};
   }
+
+  mlir::Value createLenTrim(Char str) {
+    // Note: Runtime for LEN_TRIM should also be available at some
+    // point. For now use an inlined implementation.
+    auto indexType = builder.getIndexType();
+    mlir::Value len = builder.createHere<fir::ConvertOp>(indexType, str.len);
+    auto data = materializeValue(str);
```

Comment:
Isn't `str.data` already a memory reference?

Reply:
It is. I removed that, thanks.